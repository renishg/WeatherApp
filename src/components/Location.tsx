import moment from 'moment';
import React, {FC} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {colors} from '../theme/colors';
import {fonts} from '../theme/fonts';

type LocationProps = {
  name: string;
  region: string;
  country: string;
  lastUpdated: string;
};

const Location: FC<LocationProps> = ({name, region, country, lastUpdated}) => {
  return (
    <View style={styles.container}>
      <Text testID={'name'} style={styles.nameText}>
        {name}
      </Text>
      <Text testID={'region'} style={styles.regionText}>
        {`${region}, ${country}`}
      </Text>
      <Text testID={'updatedText'} style={styles.updatedText}>
        {`Updated ${moment(lastUpdated).fromNow()}`}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
  },
  nameText: {
    color: colors.white,
    fontSize: 30,
    fontFamily: fonts.medium,
  },
  regionText: {
    color: colors.white,
    fontSize: 14,
    fontFamily: fonts.medium,
  },
  updatedText: {
    color: colors.white,
    fontSize: 16,
    fontFamily: fonts.regular,
    marginTop: 10,
  },
});

export {Location};
