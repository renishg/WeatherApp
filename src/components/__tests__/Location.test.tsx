import React from 'react';
import moment from 'moment';
import {create} from 'react-test-renderer';
import {Location} from '../Location';

describe('Location', () => {
  const mockProps = {
    country: 'United Kingdom',
    lastUpdated: '2023-03-23 18:15',
    name: 'London',
    region: 'City of London, Greater London',
  };

  it('should render the Location component correctly', () => {
    const tree = create(<Location {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the name, region, and country correctly', () => {
    const tree = create(<Location {...mockProps} />).root;
    const nameText = tree.findByProps({testID: 'name'});
    const regionText = tree.findByProps({testID: 'region'});
    expect(nameText.props.children).toBe(mockProps.name);
    expect(regionText.props.children).toBe(
      `${mockProps.region}, ${mockProps.country}`,
    );
  });

  it('should render the updated time correctly', () => {
    const tree = create(<Location {...mockProps} />).root;
    const updatedText = tree.findByProps({testID: 'updatedText'});
    expect(updatedText.props.children).toBe(
      `Updated ${moment(mockProps.lastUpdated).fromNow()}`,
    );
  });
});
