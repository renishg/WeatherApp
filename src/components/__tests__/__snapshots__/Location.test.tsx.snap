// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location should render the Location component correctly 1`] = `
<View
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-Medium",
        "fontSize": 30,
      }
    }
    testID="name"
  >
    London
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-Medium",
        "fontSize": 14,
      }
    }
    testID="region"
  >
    City of London, Greater London, United Kingdom
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
        "marginTop": 10,
      }
    }
    testID="updatedText"
  >
    Updated 20 hours ago
  </Text>
</View>
`;
